src/sync/installation.ts

```diff
@@ -209,14 +209,221 @@ export const isNotFoundError = (
+					job.sentry.setExtra("Request", {
+						host: err.request.domain,
+						path: err.request.path,
+						method: err.request.method
+					});
+				}
+
+				if (err.response) {
+					job.sentry.setExtra("Response", {
+						status: err.response.status,
+						statusText: err.response.statusText,
+						body: err.response.body
+					});
+				}
+
+				throw err;
+			}
+		}
+
+		await updateJobStatus(
+			queues,
+			job,
+			edges,
+			task,
+			repositoryId,
+			logger,
+		);
+
+		statsd.increment(metricTaskStatus.complete, [`type: ${nextTask.task}`]);
+
+	} catch (err) {
+		const rateLimit = Number(err?.headers?.["x-ratelimit-reset"]);
+		const delay = Math.max(Date.now() - rateLimit * 1000, 0);
+
+		if (delay) {
+			// if not NaN or 0
+			logger.info({ delay, job, task: nextTask }, `Delaying job for ${delay}ms`);
+			queues.installation.add(job.data, { delay });
+			return;
+		}
+
+		if (String(err).includes("connect ETIMEDOUT")) {
+			// There was a network connection issue.
+			// Add the job back to the queue with a 5 second delay
+			logger.warn({ job, task: nextTask }, "ETIMEDOUT error, retrying in 5 seconds");
+			queues.installation.add(job.data, { delay: 5000 });
+			return;
+		}
+
+		if (
+			String(err.message).includes(
+				"You have triggered an abuse detection mechanism"
+			)
+		) {
+			// Too much server processing time, wait 60 seconds and try again
+			logger.warn({ job, task: nextTask }, "Abuse detection triggered. Retrying in 60 seconds");
+			queues.installation.add(job.data, { delay: 60000 });
+			return;
+		}
+
+		// Continue sync when a 404/NOT_FOUND is returned
+		if (isNotFoundError(err, job, nextTask, logger)) {
+			const edgesLeft = []; // No edges left to process since the repository doesn't exist
+			await updateJobStatus(queues, job, edgesLeft, task, repositoryId, logger);
+			return;
+		}
+
+		if (await booleanFlag(BooleanFlags.CONTINUE_SYNC_ON_ERROR, false, jiraHost)) {
+
+			// TODO: add the jiraHost to the logger with logger.child()
+			const host = subscription.jiraHost || "none";
+			logger.warn({ job, task: nextTask, err, jiraHost: host }, "Task failed, continuing with next task");
+
+			// marking the current task as failed
+			await subscription.updateRepoSyncStateItem(nextTask.repositoryId, getStatusKey(nextTask.task as TaskType), "failed");
+
+			statsd.increment(metricTaskStatus.failed, [`type: ${nextTask.task}`]);
+
+			// queueing the job again to pick up the next task
+			queues.installation.add(job.data);
+		} else {
+
+			await subscription.update({ syncStatus: "FAILED" });
+
+			// TODO: add the jiraHost to the logger with logger.child()
+			const host = subscription.jiraHost || "none";
+			logger.warn({ job, task: nextTask, err, jiraHost: host }, "Sync failed");
+
+			job.sentry.setExtra("Installation FAILED", JSON.stringify(err, null, 2));
+			job.sentry.captureException(err);
+
+			statsd.increment(metricSyncStatus.failed);
+
+			throw err;
+		}
+	}
+}
+
 // TODO: type queues
```

Comment:
Same here we don't need to pass queues as a parameter

Reply:
If you check line 214, we are receiving it as a parameter. IMHO we should either pass it further or completely remove. Given that the purpose of this PR is to extract part of the function into a separate subfunction, I decided to pass it over, to keep PR having a single focus.